internal/services/eventgrid/eventgrid_partner_configuration_resource.go

```diff
@@ -0,0 +1,270 @@
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				existing.Model.Tags = pointer.To(config.Tags)
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, *existing.Model); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+			return nil
+		},
+	}
+}
+
+func (r EventGridPartnerConfigurationResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.PartnerConfigurations
+
+			id, err := commonids.ParseResourceGroupID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			state := EventGridPartnerConfigurationResourceModel{
+				ResourceGroup: id.ResourceGroupName,
+			}
+
+			if model := resp.Model; model != nil {
+				state.DefaultMaximumExpirationTimeInDays = *model.Properties.PartnerAuthorization.DefaultMaximumExpirationTimeInDays
+				state.PartnerAuthorizations = flattenAuthorizedPartnersList(model.Properties.PartnerAuthorization.AuthorizedPartnersList)
+
+				if model.Tags != nil {
+					state.Tags = *model.Tags
+				}
+			}
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r EventGridPartnerConfigurationResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.PartnerConfigurations
+
+			id, err := commonids.ParseResourceGroupID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+			return nil
+		},
+	}
+}
+
+func (EventGridPartnerConfigurationResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return commonids.ValidateResourceGroupID
+}
+
+func expandAuthorizedPartnersList(partnerAuthorization *[]PartnerAuthorization) *[]partnerconfigurations.Partner {
+	partners := []partnerconfigurations.Partner{}
+
+	if partnerAuthorization == nil {
+		return &partners
+	}
+
+	for _, partnerAuth := range *partnerAuthorization {
+		partners = append(partners, partnerconfigurations.Partner{
+			PartnerName:                      pointer.To(partnerAuth.PartnerName),
+			PartnerRegistrationImmutableId:   pointer.To(partnerAuth.PartnerRegistrationImmutableId),
+			AuthorizationExpirationTimeInUtc: pointer.To(partnerAuth.AuthorizationExpirationTimeInUtc),
+		})
+	}
+
+	return &partners
+}
+
+func flattenAuthorizedPartnersList(partner *[]partnerconfigurations.Partner) []PartnerAuthorization {
+	partnerAuthorizations := []PartnerAuthorization{}
```

Comment:
```suggestion
	partnerAuthorizations := make([]PartnerAuthorization, 0)
```
Literals result in `nil` if empty, and thus not allowing `append()`ing. Using `make` ensures an empty slice is returned that can accept appending.

Reply:
`append` is immune to nil, check: https://go.dev/play/p/DQHTjN7TVfG. But I have refactored this to use make just to be consistent with the other codes.